homeassistant/components/imap_email_content/sensor.py

```diff
@@ -254,22 +270,30 @@ def get_msg_text(email_message):
     def update(self) -> None:
         """Read emails and publish state change."""
         email_message = self._email_reader.read_next()
-
-        if email_message is None:
-            self._message = None
-            self._state_attributes = {}
-            return
-
-        if self.sender_allowed(email_message):
-            message = EmailContentSensor.get_msg_subject(email_message)
-
-            if self._value_template is not None:
-                message = self.render_template(email_message)
-
-            self._message = message
-            self._state_attributes = {
-                ATTR_FROM: EmailContentSensor.get_msg_sender(email_message),
-                ATTR_SUBJECT: EmailContentSensor.get_msg_subject(email_message),
-                ATTR_DATE: email_message["Date"],
-                ATTR_BODY: EmailContentSensor.get_msg_text(email_message),
-            }
+        while (
+            self._last_id is None or self._last_id != self._email_reader.last_unread_id
+        ):
+            if email_message is None:
+                self._message = None
+                self._state_attributes = {}
+                return
+
+            self._last_id = self._email_reader.last_id
+
+            if self.sender_allowed(email_message):
+                message = EmailContentSensor.get_msg_subject(email_message)
+
+                if self._value_template is not None:
+                    message = self.render_template(email_message)
+
+                self._message = message
+                self._state_attributes = {
+                    ATTR_FROM: EmailContentSensor.get_msg_sender(email_message),
+                    ATTR_SUBJECT: EmailContentSensor.get_msg_subject(email_message),
+                    ATTR_DATE: email_message["Date"],
+                    ATTR_BODY: EmailContentSensor.get_msg_text(email_message),
+                }
+
+            if self._last_id == self._email_reader.last_unread_id:
```

Comment:
How can this conditional ever be false?

https://github.com/home-assistant/core/pull/89563/files#diff-b689f8b39b04ad24757ada484bbbd5a9789f8feff07a368b7c62bb6f532393daR281

Reply:
I'll have a look at that later today.

Reply:
`self._email_reader.last_id` is the last message processed, `self._email_reader.last_unread_id` is the last message to process. The condition is false when we are at startup and we searched all messages for the last 2 days. There might be more messages that match the IMAP search, but we do not known if the sender matches.  When the last message is processed the last state set will be come the new state. If the last message processed is not the last message to process, then we read a new message until we reach the last one. To simulate this, place to matching new message in the mailbox and startup home assistant.

Reply:
Makes sense now that I'm back at my desktop.  Apparently I just needed a slightly larger monitor to see that 😉